Fix link to Symbol (#30686)
mdn/content: files/en-us/glossary/symbol/index.md

---
title: Symbol
slug: Glossary/Symbol
page-type: glossary-definition
---

{{GlossarySidebar}}

> A **symbol** is a data type that represents identifiers. Unicity is guaranteed for each of them. They are sometimes called _atoms_.

Reviewer: ```suggestion
A **symbol** is a data type that represents unique, unforgeable identifiers. They are sometimes called _atoms_.
```

> In JavaScript, `symbol` is one of the [primitives](/en-US/docs/Glossary/Primitive) and can be created using the [`Symbol()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) factory method that returns a different symbol each time. They can be used as keys for maps.

Reviewer: ```suggestion
In JavaScript, `symbol` is one of the [primitives](/en-US/docs/Glossary/Primitive) and can be created using the [`Symbol()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) factory method that returns a different symbol each time. They can be used as keys for objects which can never collide with other properties.
```

> In JavaScript, `symbol` is one of the [primitives](/en-US/docs/Glossary/Primitive) and can be created using the [`Symbol()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) factory method that returns a different symbol each time. They can be used as keys for objects which can never collide with other properties.

Reviewer: ```suggestion
In JavaScript, `symbol` is one of the [primitive types](/en-US/docs/Glossary/Primitive) and can be created using the [`Symbol()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) factory method that returns a different symbol each time. They can be used as keys for objects which can never accidentally collide with other properties.

JavaScript also defines two other categories of symbols: well-known symbols and registered symbols. Read the {{jsxref("Symbol")}} reference for more information.
```

---
title: Symbol
slug: Glossary/Symbol
page-type: glossary-definition
---

{{GlossarySidebar}}

> A **symbol** is a data type that represents unique, unforgeable identifiers. They are sometimes called _atoms_.

Reviewer 2: Why do we care? I mean why is an unforgeable identifier useful?

Reviewer: If you have an object and you want to access a property (either read it or modify it), you cannot accidentally read a property that was not what you intended unless you have the reference to the original identifier.

```js
const obj = {
  [Symbol()]: someSecret;
}

function getProperty(x) {
  return obj[x];
}

getProperty("whatever you put here"); // Will never be someSecret
```

Reviewer 2: Might be worth saying that. Something like (I might not be using correct terminology):

```suggestion
A **symbol** is a data type that represents unique, unforgeable identifiers. They are sometimes called _atoms_.

Because a symbol is unique and unforgeable, you can only read a property value associated with a symbol if you have a reference to the original identifier.
```
